src/services/server/ServerService.ts

```diff
@@ -39,6 +57,27 @@ export class ServerService {
     });
   }
 
+  getServer(serverAddress: IServerAddress): ng.IPromise<IServer> {
+    let deferred: ng.IDeferred<IServer> = this.$q.defer<IServer>();
+    this.dmrService.readResource({
+      address: this.generateAddress(serverAddress),
+      "include-runtime": true
+    }).then(server => {
+      let serverState:string = server["server-state"].toUpperCase();
```

Comment:
I know but this call is a bit specific because I can read everything I need to return from the given server object. So we would make two calls here instead of one.